File: src/test/java/com/jos/dem/vetlog/validator/PetValidatorTest.java
package com.jos.dem.vetlog.validator;

import com.jos.dem.vetlog.command.PetCommand;
import lombok.extern.slf4j.Slf4j;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInfo;
import org.springframework.validation.Errors;

import static org.mockito.ArgumentMatchers.anyString;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.verify;

@Slf4j
class PetValidatorTest {

  private PetValidator validator = new PetValidator();

  @Test
  @DisplayName("validating birthdate")
  void shouldValidateBirthdate(TestInfo testInfo) {
    log.info("Running: {}", testInfo.getDisplayName());
    Errors errors = mock(Errors.class);
    PetCommand petCommand = new PetCommand();
    petCommand.setBirthDate("2021-01-17T00:00");
    validator.validate(petCommand, errors);
    verify(errors, never()).rejectValue(anyString(), anyString());
  }
}

File: src/test/java/com/josdem/vetlog/validator/PetValidatorTest.kt
package com.josdem.vetlog.validator

import com.josdem.vetlog.command.PetCommand
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.TestInfo
import org.mockito.kotlin.any
import org.mockito.kotlin.mock
import org.mockito.kotlin.never
import org.mockito.kotlin.verify
import org.slf4j.LoggerFactory
import org.springframework.validation.Errors
import java.time.LocalDate

internal class PetValidatorTest {
    companion object {
        private val log = LoggerFactory.getLogger(PetValidatorTest::class.java)
    }

    private val validator = PetValidator()
    private val errors: Errors = mock()

    @Test
    fun `should validate birthdate`(testInfo: TestInfo) {
        log.info(testInfo.displayName)
        val petCommand = getPetCommand("2021-01-17")
        validator.validate(petCommand, errors)
        verify(errors, never()).rejectValue(any(), any())
    }

    @Test
    fun `should validate empty birthdate`(testInfo: TestInfo) {
        log.info(testInfo.displayName)
        val petCommand = getPetCommand("")
        validator.validate(petCommand, errors)
        verify(errors, never()).rejectValue(any(), any())
    }

    @Test
    fun `should reject a birthdate`(testInfo: TestInfo) {
        log.info(testInfo.displayName)
        val futureDate = LocalDate.now().plusMonths(1).toString()
        val petCommand = getPetCommand(futureDate)
        validator.validate(petCommand, errors)
        verify(errors).rejectValue("birthDate", "pet.error.birthDate.past")
    }

    private fun getPetCommand(birthdate: String): PetCommand = PetCommand().apply { birthDate = birthdate }
}
